learn/reif_examples.dj

```diff
@@ -0,0 +1,453 @@
+# reif Cookbook 
+
+This tutorial shows some examples of using the predicates in [reif](/reif.html). 
+Load lambda, dif, clpz, and reif into your toplevel to evaluate the examples in your toplevel.
+
+If you have an improvement (such as a better example or an explanation) to this tutorial, please [submit a PR](https://github.com/mthom/scryer-prolog/pulls) or 
+[github issue](https://github.com/mthom/scryer-prolog/issues).
+
+## Introduction
+
+### Exported Predicates
+
+Here are predicates exported by reif:
+
+``` prolog
+:- module(reif, [if_/3, (=)/3, (',')/3, (;)/3, cond_t/3, dif/3,
+		         memberd_t/3, tfilter/3, tmember/2, tmember_t/3,
+		         tpartition/4]).
+```
+
+### What is a reified boolean value?  
+
+In prolog it means representing the truth of some expression or relation as a boolean value.  Unlike many languages, the 
+truthyness of a term like 'A < 10' can not be directly bound to a variable with an assignment.   This is because terms have a different meaning in prolog
+that you should already be familiar with.
+
+``` prolog
+?- A is 112, C= (A<10).  
+   A = 112, C = (112<10).
+?- 
+```
+To make C bound to a boolean value with the built in (->):
+
+``` prolog
+?- A is 2, ( A#<10 -> C=true;C=false), C, format("~s",["C holds"]).
```

Comment:
`A is 2` is a tad silly, as is the call to `C` and the `format` expression.

But more importantly, can we think of an example that uses reification meaningfully?